spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -84,7 +85,7 @@ static void validateConfigParameters(Schema inputSchema, @Nullable String featur
       throw new IllegalArgumentException("Cannot specify values for both featuresToInclude and featuresToExclude. " +
                                            "Please specify fields for one.");
     }
-    Map<String, Integer> fields = new HashMap<>();
+    Map<String, Integer> fields = new LinkedHashMap<>();
```

Comment:
In the test case, instead of using schema with more than one string types, may be we can use only 1 invalid field with string type and change `Assert.assertEquals("Features must be of type : int, double, float, long but was of type STRING for " +
                            "field tailNum.", e.getMessage());` message accordingly?